app/helpers/activity_helper.rb

```diff
@@ -15,4 +15,9 @@ def activity_back_path(activity)
     return organisation_path(activity.organisation) if activity.is_fund_level?
     return organisation_activity_path(activity.parent_activity.organisation, activity.parent_activity) if activity.is_programme_level?
   end
+
+  def user_can_edit?(user, activity)
```

Comment:
This method looks like it might be doing the same authorisation that use Pundit for elsewhere.

If this is the case then I wonder if this could be replaced with something like `policy(@activity).update?` in the view? This would keep all the authorisation logic together in Pundit. We use this quite a lot on [the organisation#show view](https://github.com/UKGovernmentBEIS/beis-report-official-development-assistance/blob/develop/app/views/staff/organisations/show.html.haml#L15).

Reply:
Thanks! I still struggle to get my head around Pundit so this is helpful!